starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,386 @@
 package ly.generalassemb.drewmahrt.tictactoe;
 
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
 
 public class GameActivity extends AppCompatActivity {
+    String winner, player1, player2;
+    int turn = 1;
+    TextView a1, b1, c1, a2, b2, c2, a3, b3, c3, playerTurn;
+    public String lastPlayedLetter = " ";
+    public String lastPlayedPos = " ";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_game);
+
+        View.OnClickListener marker = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                switch (v.getId()) {
+
+                    case R.id.textView:
+                        a1.setText(xOrY());
+                        setLastPlayedLetter(a1.getText().toString());
+                        setLastPlayedPos("al");
+                        break;
+
+                    case R.id.textView2:
+                        a2.setText(xOrY());
+                        setLastPlayedLetter(a2.getText().toString());
+                        setLastPlayedPos("a2");
```

Comment:
You're repeating this logic a bunch of times - I suggest moving to a helper method and passing in the values that vary depending on which TextView was clicked.